src/circuit/range_chip.rs

```diff
@@ -42,14 +54,14 @@ impl<N: FieldExt> RangeChip<N> {
         let range_class = meta.fixed_column();
         let tag_range_table_column = meta.lookup_table_column();
         let max_range_table_column = meta.lookup_table_column();
-        let value = meta.advice_column();
+        let values = [(); VALUE_COLUMNS].map(|_| meta.advice_column());
 
-        meta.enable_equality(value);
+        meta.enable_equality(values[0]);
 
         meta.lookup("tag range check", |meta| {
             let class = meta.query_fixed(range_class, Rotation::cur());
             let is_block_first = meta.query_fixed(block_first, Rotation::cur());
-            let v = meta.query_advice(value, Rotation::cur());
+            let v = meta.query_advice(values[VALUE_COLUMNS - 1], Rotation::cur());
```

Comment:
remove `Expression::Constant(N::one()) - is_block_first` if it is not the first col